src/rules/commentFormatRule.ts

```diff
@@ -47,7 +62,30 @@ class CommentWalker extends Lint.RuleWalker {
                 }
             }
         });
-        // no need to call super to visit the rest of the nodes, so don't call super here
+    }
+
+    public visitRegularExpressionLiteral(node: ts.Node) {
+        if (node.getStart() < node.getEnd()) {
+            // only add to the map nodes whose end comes after their start, to prevent infinite loops
+            this.tokensToSkipStartEndMap[node.getStart()] = node.getEnd();
+        }
+        super.visitRegularExpressionLiteral(node);
+    }
+
+    public visitIdentifier(node: ts.Identifier) {
+        if (node.getStart() < node.getEnd()) {
+            // only add to the map nodes whose end comes after their start, to prevent infinite loops
+            this.tokensToSkipStartEndMap[node.getStart()] = node.getEnd();
+        }
+        super.visitIdentifier(node);
+    }
+
+    public visitTemplateExpression(node: ts.TemplateExpression) {
```

Comment:
There's a lot of repetition in these 3 methods. Split out this block into a private addNode method and call it from all these methods